sync_command.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/urfave/cli/v2"
+	"github.com/vshn/appuio-odoo-adapter/odoo"
+)
+
+type syncCommand struct {
+	OdooURL      string
+	OdooUsername string
+	OdooPassword string
+	OdooDB       string
+}
+
+var syncCommandName = "sync"
+
+func newSyncCommand() *cli.Command {
+	command := &syncCommand{}
+	return &cli.Command{
+		Name:   syncCommandName,
+		Usage:  "Sync Odoo entities from APPUiO Cloud",
+		Before: command.validate,
+		Action: command.execute,
+		Flags: []cli.Flag{
+			newOdooURLFlag(&command.OdooURL),
+			newOdooDBFlag(&command.OdooDB),
+			newOdooUsernameFlag(&command.OdooUsername),
+			newOdooPasswordFlag(&command.OdooPassword),
+		},
+	}
+}
+
+func (c *syncCommand) validate(context *cli.Context) error {
+	log := AppLogger(context).WithName(syncCommandName)
+	log.V(1).Info("validating config")
+	// The `Required` property in the StringFlag already checks if it's non-empty.
+	if _, err := url.Parse(c.OdooURL); err != nil {
+		return fmt.Errorf("could not parse %q as URL value for flag %s: %s", c.OdooURL, newOdooURLFlag(nil).Name, err)
+	}
+	return nil
+}
+
+func (c *syncCommand) execute(context *cli.Context) error {
+	_ = LogMetadata(context)
+	log := AppLogger(context).WithName(syncCommandName)
+
+	client := odoo.NewClient(c.OdooURL, c.OdooDB)
+	client.SetDebugLogger(log)
```

Comment:
Any preferences regarding the debug logging here?
The `--debug` CLI flag is meant to set log level to 1 for increased verbosity. However, Logging every request and response might be too verbose, so that's why I'm using log level 2 in the client.

Would you prefer to use log level 2 in the `--debug` flag already so it includes super-verbose detailed responses, or should there be a separate/additional flag to be super-verbose?

Reply:
One (super verbose) `--debug` flag is enough for me at the moment.